.github/workflows/check-build-test.yml

```diff
@@ -0,0 +1,136 @@
+name: Check, Build, and Test
+
+on:
+  pull_request:
+  push:
+    branches: [master]
+
+jobs:
+  check:
+    name: Check
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          # FIXME: since we don't know what commit the last tag was it's safer to get entire repo so previousStableVersion resolves
+          fetch-depth: 0
+
+      # FIXME: removed in favour of fetch-depth defined in Checkout step ^. i tried an arbitrarily large fetch depth 1000 and it still didn't work.
+      #      - name: Fetch tags
+      #        run: git fetch --depth=1000 origin +refs/tags/*:refs/tags/*
+
+      - name: Checkout GitHub merge
+        if: github.event.pull_request
+        run: |-
+          git fetch origin pull/${{ github.event.pull_request.number }}/merge:scratch
+          git checkout scratch
```

Comment:
I copied this blindly. I'm not sure what the problem with the `checkout` step default behaviour of using a merge commit.

There's also this:

https://github.com/actions/checkout#checkout-pull-request-head-commit-instead-of-merge-commit

Reply:
In other repos we saw that the `checkout@v2` action didn't do what the documentation said.
The hack above was identified by @chbatey.

Reply:
It doesn't use merge commit, it is a lie, @patriknw discovered this and i re-discovered it

Reply:
Thanks.